### ISSUE TITLE
Importing certain strings causes json_encode error

### ISSUE THREAD

**Issue by balgf** [algolia/scout-extended#223]
Version in question: v1.8.0

To reproduce:
1. Setup algolia search client and  scout extended.
2. Prepare table with VARCHAR column.
3. Insert a row with text. In my case, I used this "0281E3986958"
4. Run `artisan scout:import` or `artisan scout:reimport`

Observed behavior: An error occurs `InvalidArgumentException  : json_encode error: Inf and NaN cannot be JSON encoded`

Expected behavior: Should import fine.

Findings:
This was caused by "0281E3986958" data that is treated as an exponential number and was converted into `INF` (`["column" => INF]`).
Line that is causing this is [L116](https://github.com/algolia/scout-extended/blob/v1.8.0/src/Jobs/UpdateJob.php#L116)
Tried replacing `->transform($array)` with `->toArray()` and it imported fine.

**Comment by nunomaduro** [comment]
Thanks for reporthing this issue, so far you are the first reporting this problem.

You seem to have found a nice workaround too - so If someone else reports this problem, a I will link it to this issue and re-open it.

**Comment by jakew009** [comment]
Hi @nunomaduro 

Sorry to chime back in here, but this doesn't really feel like a very good reply. Fwiw's we are paying Algolia customers.

We've identified a (pretty nasty) bug in your software whereby a text string is getting treated as a exponential number.

We have identified the exact line in your package that appears to be causing it.

I don't think forking the repo is really a solution here - surely it would be better for you to replicate the issue (which should be trivial) and then implement a fix?

**Comment by jakew009** [comment]
Looking at the code I don't really understand why you are calling Laravel's `transform` class here (https://github.com/algolia/scout-extended/blob/v1.8.0/src/Jobs/UpdateJob.php#L116) without providing it a function?
https://laravel.com/docs/5.8/collections#method-transform

The docs don't mention anything about feeding it an array?

**Comment by nunomaduro** [comment]
The transform method is a macro of the builder: https://github.com/algolia/scout-extended/blob/31209a672b14b5b06b2fd9b7ea65a346f0c30e6c/src/ScoutExtendedServiceProvider.php#L126.

**Comment by georgeboot** [comment]
We're having the exact same issue. I haven't had time to nail it down to an exact value, but my guess is that for us it's the uuid's beging mistreated.

**Comment by nunomaduro** [comment]
@georgeboot Do you have override the `toArray` method?

**Comment by georgeboot** [comment]
@nunomaduro Nope, just the `toSearchableArray()` in my models:

```php
<?php

public function toSearchableArray(): array
{
    return [
        'id' => $this->id,
        'status' => $this->status,
        'name' => $this->name,
        'sales_pitch' => $this->sales_pitch,
        'vendor' => [
            'id' => $this->vendor->id,
            'name' => $this->vendor->name,
        ],
        'venue' => [
            'id' => $this->venue->id,
            'name' => $this->venue->name,
        ],
        'start_date' => $this->start_date->timestamp,
        'end_date' => $this->end_date->timestamp ?? null,
        'performers' => $this->performers->toArray(),
    ];
}
```

Note that all IDs are string-based uuid's.

**Comment by nunomaduro** [comment]
Hummm. If you have a min, can you do a `dd` to see if the `transform` is being applied? https://github.com/algolia/scout-extended/blob/ea5cec58af8f726fe1f5848082c0a5f84ffbd3b8/src/Jobs/UpdateJob.php#L116.

It shouldn't.

**Comment by georgeboot** [comment]
> It shouldn't.

But it does :) Not for the regular searchable model, but it does for an aggreagated search (`Algolia\ScoutExtended\Searchable\Aggregator`)

**Comment by nunomaduro** [comment]
I see. Having something like this on the Aggregator Model will fix your issue:

```php
public function toSearchableArray(): array
{
    return $this->toArray();
}
```

**Comment by nunomaduro** [comment]
cc @jakew009 By default, when the `toSearchableArray` do not exists, we transform some data for you. Numeric strings to ints, dates objects to string, etc.

**Comment by jakew009** [comment]
cc @balgf did you see this?

**Comment by mitchierichie** [comment]
can you please reopen? and I'll close with my fix #360 @nunomaduro

this is a bug in this library, and there is no reason to leave it as is, a value that will end up as NaN or INF will not make it to Algolia, so why continue casting it, we could leave it as a string?
